ui/Screen/Network.svelte

```diff
@@ -0,0 +1,180 @@
+<!--
+ Copyright © 2021 The Radicle Upstream Contributors
+
+ This file is part of radicle-upstream, distributed under the GPLv3
+ with Radicle Linking Exception. For full terms see the included
+ LICENSE file.
+-->
+<script lang="typescript">
+  import { status } from "ui/src/localPeer";
+  import { indicatorState } from "ui/src/network";
+  import {
+    settings,
+    seedValidation,
+    addSeed,
+    removeSeed,
+  } from "../src/session";
+
+  import {
+    Button,
+    Icon,
+    SidebarLayout,
+    StyledCopyable,
+    TextInput,
+  } from "ui/DesignSystem";
+
+  let seedInputValue: string = "";
+
+  const submitSeed = async () => {
+    if (await addSeed(seedInputValue)) {
+      seedInputValue = "";
+    }
+  };
+
+  $: if (seedInputValue === "") {
+    seedValidation.reset();
+  }
+
+  let statusText: string = "";
+  let statusFill: string = "";
+  $: {
+    const state = indicatorState($status);
+    if (state) {
+      statusText = state.text;
+      statusFill = state.fill;
+    }
+  }
```

Comment:
There’s a more concise alternative to this. But I’m also ok with keeping it as it it.

```typescript
import * as svelteStore from "svelte/store";

const indicatorStatus = svelteStore.derived(status, indicatorState);
```
Then you can use `$indicatorStatus.text` instead of `statusText` etc.

This depends on fixing the return type of `indicatorState` as mentioned in the comment above.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/2066/commits/841cb0998cff274d0adfc34bb2b6b6bb458a8df9